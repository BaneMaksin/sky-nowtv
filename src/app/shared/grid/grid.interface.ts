/*
 * Date: 7/15/22, 8:33 AM
 * Copyright: Branislav Maksin (c) 2022
 */

import { ReactNode } from 'react';

// Grip component input properties
export interface GridProps {
  children: ReactNode;
  className?: string;
}
