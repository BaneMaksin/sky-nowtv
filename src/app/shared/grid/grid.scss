/*!
 * Date: 7/15/22, 8:33 AM
 * Copyright: Branislav Maksin (c) 2022
 */

.wrapper {
  grid-template-columns: repeat(auto-fill, minmax(294px, auto));
  grid-gap: 20px;
  display: grid;
}
