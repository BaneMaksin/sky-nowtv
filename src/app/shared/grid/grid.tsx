/*
 * Date: 7/15/22, 8:33 AM
 * Copyright: Branislav Maksin (c) 2022
 */

import React from 'react';
import cn from 'classnames';

import { GridProps } from './grid.interface';

// Style
import style from './grid.scss';

/**
 * Shared grid stateless component.
 *
 * @param className - Style class.
 * @param children - Children elements.
 * @returns Virtual element.
 */
export const Grid = ({ className, children }: GridProps): JSX.Element => (
  <div
    className={cn(
      style.wrapper,
      { [className]: !!className }
    )}
  >
    {children}
  </div>
);
