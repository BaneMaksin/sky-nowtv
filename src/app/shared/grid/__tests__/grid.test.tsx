/*
 * Date: 7/15/22, 8:33 AM
 * Copyright: Branislav Maksin (c) 2022
 */

import React from 'react';
import { render } from '@testing-library/react';

import { Grid } from '../grid';

describe('Shared grid component', () => {

  it('should render grid component without custom style class', () => {

    // Render the component
    const { asFragment } = render(
      <Grid>
        Foo Bar
      </Grid>
    );

    // Match the snapshot
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render grid component with custom style class', () => {

    // Render the component
    const { container } = render(
      <Grid className="foo-bar">
        Foo Bar
      </Grid>
    );

    // Find the custom class
    expect(container.querySelector('div').classList.contains('foo-bar'))
      .toBeTruthy();
  });
});
