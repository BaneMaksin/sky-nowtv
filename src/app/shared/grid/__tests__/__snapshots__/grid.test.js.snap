// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared grid component should render grid component without custom style class 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    Foo Bar
  </div>
</DocumentFragment>
`;
